### PR TITLE
Correct torch.nn.CrossEntropyLoss output shape specification

### LINKED ISSUE #79531: torch.nn.CrossEntropyLoss output shape specification does not match behaviour
### 📚 The doc issue

https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html#torch.nn.CrossEntropyLoss

Just slightly changing the provided example we can see
```
loss = nn.CrossEntropyLoss(reduction = "none")
# Example of target with class probabilities
input = torch.randn(3, 5, requires_grad=True)
target = torch.randn(3, 5).softmax(dim=1)
output = loss(input, target)

target.shape
# torch.Size([3, 5])
output.shape
# torch.Size([3])
```

This does not match the provided shape specification that 

`Output: If reduction is ‘none’, same shape as the target. Otherwise, scalar.`

### Suggest a potential alternative/fix

`Output: If target contains probabilities and reduction is ‘none’, shape (N). If target contains class indices and reduction is 'none', same shape as target. Otherwise scalar.`

cc @svekars @holly1238 @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are

### PR DESCRIPTION
**Pull request by azhx**
Fixes #79531

### CODE REVIEW

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -1125,7 +1125,9 @@ class probabilities only when a single class label per minibatch item is too res
         - Target: If containing class indices, shape :math:`()`, :math:`(N)` or :math:`(N, d_1, d_2, ..., d_K)` with
           :math:`K \geq 1` in the case of K-dimensional loss where each value should be between :math:`[0, C)`.
           If containing class probabilities, same shape as the input and each value should be between :math:`[0, 1]`.
-        - Output: If reduction is 'none', same shape as the target. Otherwise, scalar.
+        - Output: If reduction is ‘none’, shape (), (N) or (N, d_1, d_2, ..., d_K) with K≥1
+          in the case of K-dimensional loss, depending on the shape of the input. Otherwise, scalar.
```
Comment (jbschlosser) [inline torch/nn/modules/loss.py:1129]:
```suggestion
        - Output: If reduction is 'none', shape :math:`()`, :math:`(N)` or :math:`(N, d_1, d_2, ..., d_K)` with :math:`K \geq 1`
          in the case of K-dimensional loss, depending on the shape of the input. Otherwise, scalar.
```

**Review by jbschlosser — COMMENTED** [review]
Thanks for the update! Small formatting nit, otherwise LGTM

**Review by jbschlosser — APPROVED** [review]
LGTM!

**Comment by azhx** [comment]
> Merge failed due to Refusing to merge as mandatory check(s) [Facebook CLA Check](https://code.intern.facebook.com/cla/) failed for rule superuser Raised by https://github.com/pytorch/pytorch/actions/runs/2498625541

I submitted my CLA right after I opened the PR, not sure when that check is gonna be rerun
